File: DeepSpace-Java/src/deepspace/BetaPowerEfficientSpaceStation.java
package deepspace;

public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation {
    private float EXTRAEFFICIENCY = 1.2f;

    private Dice dice = new Dice();


    BetaPowerEfficientSpaceStation(SpaceStation estacion) {
        super(estacion);
    }

    @Override
    public float fire() {
        return dice.extraEfficiency()
            ?   super.fire() * EXTRAEFFICIENCY
            :   super.fire();
    }

    public BetaPowerEfficientSpaceStationToUI getUIversion () {
        return new BetaPowerEfficientSpaceStationToUI(this);
    }

    public String toString() {
        return "Beta " + super.toString();
    }
}

File: DeepSpace-Ruby/lib/BetaPowerEfficientSpaceStation.rb
# encoding:utf-8

require_relative "Dice"
require_relative "BetaPowerEfficientSpaceStationToUI"

module Deepspace
    class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
        @@EXTRAEFFICIENCY = 1.2
        
        def initialize (estation)
            super
            @dice = Dice.new
        end

        def fire
            if @dice.extraEfficiency
                disparo = super*@@EXTRAEFFICIENCY
            else
                super
            end
        end

        def getUIversion
            BetaPowerEfficientSpaceStationToUI.new(self)
        end

        def to_s
            getUIversion.to_s
        end

    end

end
